config_test.go

```diff
@@ -0,0 +1,255 @@
+package main
+
+// nolint: lll
+import (
+	"io/ioutil"
+	"net"
+	"os"
+	"testing"
+
+	"github.com/brigadecore/brigade-bitbucket-gateway/internal/webhooks"
+	"github.com/brigadecore/brigade-foundations/http"
+	"github.com/brigadecore/brigade/sdk/v2/restmachinery"
+	"github.com/stretchr/testify/require"
+)
+
+// Note that unit testing in Go does NOT clear environment variables between
+// tests, which can sometimes be a pain, but it's fine here-- so each of these
+// test functions uses a series of test cases that cumulatively build upon one
+// another.
+
+func TestAPIClientConfig(t *testing.T) {
+	testCases := []struct {
+		name       string
+		setup      func()
+		assertions func(
+			address string,
+			token string,
+			opts restmachinery.APIClientOptions,
+			err error,
+		)
+	}{
+		{
+			name:  "API_ADDRESS not set",
+			setup: func() {},
+			assertions: func(
+				_ string,
+				_ string,
+				_ restmachinery.APIClientOptions,
+				err error,
+			) {
+				require.Error(t, err)
+				require.Contains(t, err.Error(), "value not found for")
+				require.Contains(t, err.Error(), "API_ADDRESS")
+			},
+		},
+		{
+			name: "API_TOKEN not set",
+			setup: func() {
+				os.Setenv("API_ADDRESS", "foo")
+			},
+			assertions: func(
+				_ string,
+				_ string,
+				_ restmachinery.APIClientOptions,
+				err error,
+			) {
+				require.Error(t, err)
+				require.Contains(t, err.Error(), "value not found for")
+				require.Contains(t, err.Error(), "API_TOKEN")
+			},
+		},
+		{
+			name: "SUCCESS not set",
```

Comment:
nit: perhaps just `"success"` as the name?

Reply:
How do I keep doing that?